key.js

```diff
@@ -1,18 +1,321 @@
+        });
+    }
+    function decrementGameLife() {
+        gameState.lives--;
+    }
+
+    function printGameState() {
+        console.log(gameState);
+    }
+
+    return {
+        startGame,
+        gameOver,
+        levelUp,
+        resetGameLivesAndTimer,
+        resetGameState,
+        muteAudio,
+        unMuteAudio,
+        decrementGameLife,
+        printGameState,
+        getGameLife,
+        endGame,
+        gameMusicPlay,
+        gameMusicStop
+    }
+    // I. create a game state object (ensure it's a deep copy of initialGameState)
+    // II. create utility methods to manipulate the game state object
+    // 1. incrementGameLevel
+    // 2. getGameLevel
+    // 3. startGame
+    // 4. endGame
+    // 5. isGameRunning
+    // 6. isGameOver
+    // 7. getGameLife
+    // 8. decrementGameLife
+    // 9. resetGameState
+    // 10. getGameAudio
+    // 11. printGameState // just for debugging
+    // 12. <create any other method you feel that makes sense>
+    // III. return an object with all the aforementioned utility methods from 1..9.
+
+    // return {
+    //     incrementGameLevel: () => gameState.levelsInfo.currentLevel++
+    // }
+})();
+// IV. Refactor the code below to use the gameModule above for creating the game logic.
 // Button Animation Logic :
 
+const viewModule = (function () {
+    const DOMStrings = {
+        entireContainer: 'container',
+        changeHeading: 'main_heading',
+        defaultLives: 'default',
+        currentLives: 'lives',
+        mainButton: 'main',
+        liveDisplay: 'heart',
+        timerDisplay: 'timer',
+        keyDirections: 'btns',
+        randomKeyArrows: 'arrows',
+        keyboardContents: 'contents'
+    }
+    const newSequence = () => {
+        const arr = [];
+        let i = 10;
+        while (i--) {
+            const random = Math.random();
+            if (random <= .25) {
+                arr.push("&uarr;");
+            }
+            else if (random <= .50) {
+                arr.push("&rarr;");
+            }
+            else if (random <= .75) {
+                arr.push("&rarr;");
+            }
+            else {
+                arr.push("&darr;");
+            }
+        }
+        span.forEach((item, i) => {
+            item.innerHTML = arr[i];
+            if (Math.random() <= Math.random()) {
+                item.style.color = 'green';
+            } else {
+                item.style.color = 'red';
+            } i++;
+        })
+    }
+    function animateToLevelZero() {
+        defaultLife.textContent = `  x ${gameState.lives}`;
+        heart.style.display = ``;
+        timerElement.style.display = ``;
+        startButton.style.fontSize = `45px`
+        timer = setInterval(() => {
+            gameState.seconds = gameState.seconds - 1;
+            timerElement.textContent = `${gameState.seconds} `;
+            if (gameState.seconds <= 0) {
+                clearInterval(timer);
+            }
+        }, 1000)
```

Comment:
Call this as a simple method:

```suggestion
        startGameTime(timer);
```

```js
// Somewhere in the viewModule:

function startGameTime(intervalObject) {
  // Note: intervalObject is used to assign the `setInterval`'s return value to `clearInterval` later.  So be careful with this object
  // same logic as what you've already written before for `setInterval`.
}
```